e2e/kubectl.e2e.spec.ts

```diff
@@ -0,0 +1,117 @@
+import path from 'path';
+import {
+  ElectronApplication, BrowserContext, _electron, Page, Locator
+} from 'playwright';
+import { test, expect } from '@playwright/test';
+import { TestUtils } from './utils/TestUtils';
+import * as tools from './utils/ToolsUtils';
+
+let page: Page;
+const defaultReportFolder = path.join(__dirname, 'reports/');
+
+test.describe.serial('Rancher Desktop - K8s Deploy Test', () => {
+  let mainTitle: Locator;
+  let utils: TestUtils;
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  const mainTitleSelector = '[data-test="mainTitle"]';
+
+  test.beforeAll(async() => {
+    utils = new TestUtils();
+    utils.createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        '--disable-dev-shm-usage',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: `${ defaultReportFolder }pw-trace.zip` });
+    await electronApp.close();
+  });
+
+  test('should load Rancher Desktop App', async() => {
+    mainTitle = page.locator(mainTitleSelector);
+
+    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+  });
+
+  test('should start loading the background services', async() => {
+    const progressBarSelector = page.locator('.progress');
+
+    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
+    await expect(progressBarSelector).toBeHidden();
+  });
+
+  test('should run Kubernetes on Rancher Desktop (kubectl)', async() => {
+    // CI take longer than local env to start up
+    // giving it some time to proper start the background services
+    if (process.env.CI) {
+      console.log('Waiting for services - CI take a while to load...');
+      await utils.delay(200_000);
+    }
+    const output = await tools.kubectl('cluster-info');
+
+    expect(output).toMatch(/is running at ./);
+  });
+
+  test('should create a sample namespace', async() => {
+    const getExistingNamespaces = (await tools.kubectl('get', 'namespace', '--output=name')).trim();
+
+    try {
+      if (getExistingNamespaces.includes('rd-nginx-demo')) {
+        await tools.kubectl('delete', 'ns', 'rd-nginx-demo');
```

Comment:
```suggestion
        await tools.kubectl('delete', 'namespace', 'rd-nginx-demo');
```
Similar to short options, please use full subcommands in scripts / tests.

```suggestion
        await tools.kubectl('delete', '--ignore-not-found', 'namespace', 'rd-nginx-demo');
```
If you tell it to ignore if the item is not found, we don't need the check to see if it exists.  This _technically_ avoids a time of check to time of use race, but I don't think it matters here (but still nice to have less code).